app/lib/pre-apply-handler.js

```diff
@@ -0,0 +1,27 @@
+import { CLAIM_STATUS } from "ffc-ahwr-common-library";
+import { getLatestApplicationsBySbi } from "../api-requests/application-api.js";
+import { applicationType } from "../constants/constants.js";
+import { getFarmerApplyData } from "../session/index.js";
+import { keys } from "../session/keys.js";
+
+const { organisation: organisationKey } = keys.farmerApplyData;
+
+export const preApplyHandler = async (request, h) => {
+  if (request.method === "get") {
+    const organisation = getFarmerApplyData(request, organisationKey);
+
+    request.logger.setBindings({ sbi: organisation.sbi });
+
+    const latestApplications = await getLatestApplicationsBySbi(organisation.sbi);
```

Comment:
Slightly concerned with this, as will mean we have to complete a full round trip from apply -> backend -> database for every hit of any of the handlers that use this pre-handler. Can latest applications not be fetched from remote once and kept in session?